src/neuroglancer/datasource/graphene/backend.ts

```diff
@@ -0,0 +1,325 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {decodeGzip} from 'neuroglancer/async_computation/decode_gzip_request';
+import {requestAsyncComputation} from 'neuroglancer/async_computation/request';
+import {ChunkManager, WithParameters} from 'neuroglancer/chunk_manager/backend';
+import {GenericSharedDataSource} from 'neuroglancer/chunk_manager/generic_file_source';
+import {WithSharedCredentialsProviderCounterpart} from 'neuroglancer/credentials_provider/shared_counterpart';
+import {assignMeshFragmentData, FragmentChunk, ManifestChunk, MeshSource} from 'neuroglancer/mesh/backend';
+import {fetchSpecialHttpByteRange, getGrapheneFragmentKey, responseIdentity} from 'neuroglancer/datasource/graphene/base';
+import {CancellationToken} from 'neuroglancer/util/cancellation';
+import {Borrowed} from 'neuroglancer/util/disposable';
+import {convertEndian32, Endianness} from 'neuroglancer/util/endian';
+import {cancellableFetchOk, isNotFoundError, responseArrayBuffer, responseJson} from 'neuroglancer/util/http_request';
+import {stableStringify} from 'neuroglancer/util/json';
+import {getObjectId} from 'neuroglancer/util/object_id';
+import {SpecialProtocolCredentials, SpecialProtocolCredentialsProvider} from 'neuroglancer/util/special_protocol_request';
+import {cancellableFetchSpecialOk} from 'neuroglancer/datasource/graphene/base';
+import {Uint64} from 'neuroglancer/util/uint64';
+import {registerSharedObject} from 'neuroglancer/worker_rpc';
+import {ChunkedGraphSourceParameters, MeshSourceParameters} from 'neuroglancer/datasource/graphene/base';
+import {DataEncoding, ShardingParameters} from 'neuroglancer/datasource/precomputed/base';
+import {ChunkedGraphChunk, ChunkedGraphChunkSource, decodeSupervoxelArray} from 'neuroglancer/sliceview/chunked_graph/backend';
+import {DecodedMinishardIndex, decodeManifestChunk} from 'neuroglancer/datasource/precomputed/backend';
+
+interface MinishardIndexSource extends
+    GenericSharedDataSource<string, DecodedMinishardIndex|undefined> {
+  sharding: ShardingParameters;
+  credentialsProvider: SpecialProtocolCredentialsProvider;
+}
+
+function getMinishardIndexDataSource(
+    chunkManager: Borrowed<ChunkManager>, credentialsProvider: SpecialProtocolCredentialsProvider,
+    parameters: {url: string, sharding: ShardingParameters|undefined, layer:number}): MinishardIndexSource|
+    undefined {
+  const {url, sharding, layer} = parameters;
+  if (sharding === undefined) return undefined;
+  const source =
+      GenericSharedDataSource.get<string, DecodedMinishardIndex|undefined>(
+          chunkManager, stableStringify({
+            type: 'graphene:shardedDataSource',
+            url,
+            sharding,
+            credentialsProvider: getObjectId(credentialsProvider),
+          }),
+          {
+            download: async function(
```

Comment:
I think this logic may be identical to what is in the precomputed data source.  If so it would be better to refactor to avoid duplication.

Reply:
I removed this since I realized it was only being by unverified sharded meshes. I'm still working on a good explanation of verify/unverified meshes, essentially our chunked graph can give us fast but unreliable meshes (unverified) which may have gaps/holes. We are using it immediately after proofreading to provide more real time feedback. Since I removed the proofreading logic from this PR, I also stripped out the unverified mesh logic.